addon/components/hyper-table/views.js

```diff
@@ -1,12 +1,38 @@
 import Component from '@ember/component';
+import { observer } from '@ember/object';
+import { run } from '@ember/runloop';
 
 export default Component.extend({
   isHiddenAddViewModal: true,
   isHiddenUpdateViewModal: true,
   newViewName: '',
   selectedView: null,
+  filteredViews: null,
+
+  _searchQuery: '',
+  
+
+  _searchQueryObserver: observer('_searchQuery', function() {
+    run.debounce(this, () => {
+      if (this._searchQuery !== null) {
+        this.set('filteredViews', this.manager.views.filter((view) => {
+          return view.name.toLowerCase().includes(this._searchQuery.toLowerCase());
+        }));
+        return;
+      }
+
+      this.set('filteredViews', null)
+    }, 500);
```

Comment:
Can you check, but I'm pretty sure this will not debounce properly since you use arrow function. You should use a named function to ensure the good behavior